File: src/main/java/com/abhinav/lms/review/service/ReviewService.java
package com.abhinav.lms.review.service;

import com.abhinav.lms.review.dto.CourseReviewsSummaryResponse;
import com.abhinav.lms.review.dto.ReviewRequest;
import com.abhinav.lms.review.dto.ReviewResponse;
import com.abhinav.lms.security.model.UserPrincipal;

import java.util.UUID;

public interface ReviewService {

    ReviewResponse createReview(UUID courseId, ReviewRequest request, UserPrincipal currentUser);

    ReviewResponse updateReview(UUID courseId, UUID reviewId, ReviewRequest request, UserPrincipal currentUser);

    void deleteReview(UUID courseId, UUID reviewId, UserPrincipal currentUser);

    CourseReviewsSummaryResponse getCourseReviews(UUID courseId, int page, int size);
}

File: lms-frontend/services/reviewService.js
import api from './api';

export const reviewService = {
  async getCourseReviews(courseId, page = 0, size = 10) {
    const response = await api.get(`/courses/${courseId}/reviews`, {
      params: { page, size }
    });
    return response.data;
  },

  async createReview(courseId, reviewData) {
    const response = await api.post(`/courses/${courseId}/reviews`, reviewData);
    return response.data;
  },

  async updateReview(courseId, reviewId, reviewData) {
    const response = await api.put(`/courses/${courseId}/reviews/${reviewId}`, reviewData);
    return response.data;
  },

  async deleteReview(courseId, reviewId) {
    await api.delete(`/courses/${courseId}/reviews/${reviewId}`);
  }
};

export default reviewService;
